### ISSUE TITLE
Implement Vision Language Model chapter

### ISSUE THREAD

**Issue by burtenshaw** [huggingface/smol-course#49]
We need to implement the section on VLMs. It should be based on existing content from the huggingface ecosystem, adapted for SmolVLM, adapted to the course structure, and offer exercises. 

## Material

- https://github.com/huggingface/cookbook/pull/246
- https://huggingface.co/learn/cookbook/fine_tuning_vlm_trl
- https://github.com/merveenoyan/smol-vision/blob/main/Smol_VLM_FT.ipynb
- https://github.com/huggingface/smollm/blob/main/finetuning/Smol_VLM_FT.ipynb
- https://huggingface.co/blog/smolvlm
- https://huggingface.co/blog/vlms

## Steps to do

- [ ] add prose that explains vlm in simplest possible terms
- [ ] add references to all pages
- [ ] add exercise notebook on finetuning SmolVLM

**Comment by duydl** [comment]
Hi, could the issue be assigned?

**Comment by burtenshaw** [comment]
> Hi, could the issue be assigned?

Of course, I expect others will contribute too, but open a draft PR with the outline and LGTM.

**Comment by jungnerd** [comment]
Hi there 👋🏻 
I'd like to contribute to this issue by providing an exercise notebook demonstrating how to fine-tune VLM with TRL?
The notebook will be based on [How to Fine-Tune Multimodal Models or VLMs with Hugging Face TRL](https://www.philschmid.de/fine-tune-multimodal-llms-with-trl) but using a SmolVLM instead.

**Comment by burtenshaw** [comment]
That's great @jungnerd . Thanks.

We're already working on the text here: https://github.com/huggingface/smol-course/pull/59 . There's a first draft of the notebook from @duydl .

Why don't Open a PR on to https://github.com/huggingface/smol-course/pull/59 with an exercise notebook?

**Comment by duydl** [comment]
@jungnerd I had been working on sft for vlm and just finished it. There is still the dpo approach notebook need to be done [Hugging Face Blog: Preference Optimization for VLMs](https://huggingface.co/blog/dpo_vlm) and probably lots of minor fixes in completed the markdown and notebooks for vlm.

**Comment by jungnerd** [comment]
@duydl Am I understanding correctly that you’re suggesting creating a notebook to fine-tune SmolVLM using DPO? If so,  that sounds great! I think it would be really exciting to work on a fine-tuning notebook with DPO!
